### PR TITLE
Add reading message type in Fetch

### PR DESCRIPTION
**Pull request by wkal-pubnub**

### CODE REVIEW

File: src/main/java/com/pubnub/api/models/consumer/history/HistoryMessageType.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.pubnub.api.models.consumer.history;
+
+import com.pubnub.api.PubNubException;
+
+public enum HistoryMessageType {
+    Message,
```
Comment (marcin-cebo) [inline src/main/java/com/pubnub/api/models/consumer/history/HistoryMessageType.java:6]: I would change constant value to uppercase i.e. Message to MESSAGE and File to FILE.
I know that in kotlin we have them like this but I guess uppercase for constant values in enum is common standard.

We could create story to change them in kotlin as well and release in one of breaking change release?

File: src/main/java/com/pubnub/api/models/consumer/history/PNFetchMessageItem.java [hunk 2]
```diff
@@ -30,12 +31,13 @@ public class PNFetchMessageItem {
...
-    private final String messageType;
-    private int getMessageType() {
-        if (messageType == null || messageType.isEmpty()) {
-            return 0;
-        } else {
-            return Integer.parseInt(messageType);
+    private final Integer messageType;
+
+    public HistoryMessageType getMessageType() {
+        try {
```
Comment (marcin-cebo) [inline src/main/java/com/pubnub/api/models/consumer/history/PNFetchMessageItem.java:37]: I wrote a test:

```
@Test
public void testFetchSingleChannel_includeMessageTypeIsFalse() throws PubNubException {
    final String expectedChannelName = random();

    publishMixed(pubNub, 10, expectedChannelName);

    final PNFetchMessagesResult fetchMessagesResult = pubNub.fetchMessages()
            .channels(Collections.singletonList(expectedChannelName))
            .includeMessageType(false)
            .sync();

    pause(3);

    assert fetchMessagesResult != null;
    for (PNFetchMessageItem messageItem : fetchMessagesResult.getChannels().get(expectedChannelName)) {
        assertEquals(HistoryMessageType.Message ,messageItem.getMessageType());
    }

}
```

The thing is that following assertion passes:
`assertEquals(HistoryMessageType.Message ,messageItem.getMessageType());`
but in test there is `.includeMessageType(false)`
I would expect messageItem.getMessageType() to return null in this case ?


We could add to PNFetchMessageItem class:
```
    @Getter(AccessLevel.PRIVATE)
    private final boolean includeMessageType;
```

and then something like:

```
    public HistoryMessageType getMessageType() {
        return includeMessageType ? HistoryMessageType.of(messageType) : null;
    }
```

What do you think?

Comment (wkal-pubnub) [reply src/main/java/com/pubnub/api/models/consumer/history/PNFetchMessageItem.java:45]: you're right! good catch. fixed

File: src/main/java/com/pubnub/api/models/consumer/history/HistoryMessageType.java (hunk 1 above)
Comment (wkal-pubnub) [reply src/main/java/com/pubnub/api/models/consumer/history/HistoryMessageType.java]: yeah, I we should probably change both to uppercase

**Review by marcin-cebo — APPROVED** [review]

**Pull request merged**